ecommerce/taxes/lib/taxes/product.rb

```diff
@@ -8,15 +8,15 @@ def initialize(id)
       @id = id
     end
 
-    def set_vat_rate(vat_rate)
-      raise VatRateNotApplicable unless vat_rate_applicable?(vat_rate)
+    def set_vat_rate(vat_rate, catalog)
+      raise VatRateNotApplicable unless vat_rate_applicable?(vat_rate.code, catalog)
```

Comment:
The validation checking if the vat rate is available should happen outside of the domain model

Reply:
Thanks @lukaszreszke! Can you help me figure out where to put it? Is command handler a better place for it?

Reply:
Yes - command handler is better place for checking such a logic.

Reply:
Done!